canonical-web-and-design/maas-ui
# fix(ui): use resources endpoints for pod and vmcluster storage pools

## Done

- Replaced the use of `pod.storage_pools` in UI components with `pod.resources.storage_pools`, which includes `allocated_other` data. The "other" data has been included in the KVM list in this PR.
- Removed the cluster storage pool selector, which was incorrectly aggregating each cluster member's pools. Instead it's been replaced with `vmcluster.total_resources.storage_pools`, which takes a similar (but not identical) shape to `pod.resources.storage_pools`.
- Updated components to be able to work with either pod or vmcluster resource.storage_pools
- Since the new data shape does not include the pod id, I've had to remove the "default" label and sorting for now. https://bugs.launchpad.net/maas/+bug/1949681

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- In the KVM list, hover over a host that uses "other" storage and check that it is reflected in the storage popover.
- Go to the details page of a LXD KVM that uses "other" storage.
- Check that the "Storage" section in the VMs card shows the correct data (for now, allocated is the sum of "tracked" and "other". Next step will be to reflect "other" data in the details pages separately).
- Click "Add virtual machine" and scroll down to the storage table
- Click the location dropdown and check that the "Allocated" amount is the sum of "tracked" and "other"
- Close the form and go to the "Resources" tab
- Check that the storage cards show "Allocated" as the sum of "tracked" and "other"

## Fixes

Fixes canonical-web-and-design/app-squad#487
